litestar/config/response_cache.py

```diff
@@ -35,6 +41,22 @@ def default_cache_key_builder(request: Request[Any, Any, Any]) -> str:
     return request.url.path + urlencode(query_params, doseq=True)
 
 
+def default_do_cache_predicate(_: HTTPScope, status_code: int) -> bool:
+    """Given a status code, returns a boolean indicating whether the response should be cached.
+
+    Args:
+        _: ASGI scope.
+        status_code: status code of the response.
+
+    Returns:
+        A boolean indicating whether the response should be cached.
+    """
+    return HTTP_200_OK <= status_code < HTTP_300_MULTIPLE_CHOICES or status_code in (
+        HTTP_301_MOVED_PERMANENTLY,
+        HTTP_308_PERMANENT_REDIRECT,
+    )
```

Comment:
Any thoughts on what a default for this should be?

Reply:
I'm not sure what you mean by default, but these specific status codes conform to the the [PRG pattern](https://en.wikipedia.org/wiki/Post/Redirect/Get)

Reply:
I suppose I just mean, what responses should we not cache by default. E.g., this will cache a 201 CREATED which I imagine wouldn't be common.

But then again, a user would have to set `cache=True` on a POST route for this to occur, and at that point, they are just getting what they ask for I guess.

Reply:
`201`, `202`, `204`, `206`, `304`? Not really sure the right choice here for defaults though..

Reply:
One thing to consider is that where caching is enabled, we've been caching _everything_. While we obviously don't want to cache error responses, the success responses like 201 - it is not so clear to me that we can or should change that behavior. I just know I can be a bit conservative on this type of stuff, so IDK..

Reply:
The caching of everything has been a source of bugs though. I think the current default you have is fine, but we should also exclude caching of responses to "OPTIONS" requests. Then maybe in v3 we could just go ahead and only cache responses to "GET" requests by default?

Reply:
> Then maybe in v3 we could just go ahead and only cache responses to "GET" requests by default?

I don't think that's needed, as you can already selectively enable/disable the caching per route handler, which also means per method. This is only an issue for `OPTIONS` because it doesn't necessarily have an explicit route handler.

However, there is another bug at play here which is causing issues; We don't take the request method into account when calculating the cache key. This means that, not just for `OPTIONS` requests, requests that go to the same route will overwrite each other's caches.

Reply:
> you can already selectively enable/disable the caching per route handler, which also means per method

Makes sense to me.

> We don't take the request method into account when calculating the cache key

Aah you're right.

Reply:
> We don't take the request method into account when calculating the cache key.

Yeh this is prob the best solution to #2573

Reply:
I'll leave this as is for now then.